internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -543,25 +543,23 @@ func (c *enrollCmd) enrollWithBackoff(ctx context.Context, persistentConfig map[
 	defer close(signal)
 	backExp := c.backoffFactory(signal)
 
+RETRYLOOP:
 	for {
-		retry := false
 		switch {
 		case errors.Is(err, fleetapi.ErrTooManyRequests):
 			c.log.Warn("Too many requests on the remote server, will retry in a moment.")
-			retry = true
 		case errors.Is(err, fleetapi.ErrConnRefused):
 			c.log.Warn("Remote server is not ready to accept connections(Connection Refused), will retry in a moment.")
-			retry = true
 		case errors.Is(err, fleetapi.ErrTemporaryServerError):
 			c.log.Warnf("Remote server failed to handle the request(%s), will retry in a moment.", err.Error())
-			retry = true
+		case errors.Is(err, context.Canceled), errors.Is(err, context.DeadlineExceeded), err == nil:
+			break RETRYLOOP
 		case err != nil:
-			c.log.Warnf("Enrollment failed: %s", err.Error())
+			c.log.Warnf("Error detected: %s, will retry in a moment.", err.Error())
 		}
-		if !retry {
+		if !backExp.Wait() {
```

Comment:
I believe this returns false when the context is cancelled, should this not `break RETRYLOOP`?

Reply:
The label on break isn't necessary here, but I think it might be better for readability to have it anyway. Does the linter complain about it?

As for the context, we don't stop the wait on cancellation, but I think we should? We don't do anything useful with the signalling channel, so may as well use `ctx.Done()` here I think.

Reply:
I've added the label for readability.
Context cancellation is checked is a case above (the  `c.enroll()` method will return a context related error if there is cancellation/timeout).

Reply:
If the context is cancelled while we're in `backexp.Wait()`, we won't actually exit from it, and I think we should. Is that not right?

Reply:
correct, but that's a limitation with the backoff; I think it should be changed in a follow-up PR